lambdas/services/im_alerting_service.py

```diff
@@ -416,8 +412,17 @@ def unpack_alarm_history_emoji_name(
     def create_action_url(self, base_url: str, alarm_name: str) -> str:
         search_query = "#:~:text="
 
-        url_extension = alarm_name.partition("_")[2]  # remove the "<ENV>_" prefix
-        url_extension.replace(" ", "%20")
+        prefixes = [f"{os.environ['WORKSPACE']}_", f"{os.environ['WORKSPACE']}-"]
+
+        for prefix in prefixes:
+            alarm_name = alarm_name.replace(prefix, "")
+
+        url_extension = (
+            alarm_name.replace(" ", "%20")
+            .replace("-", "%20")
+            .replace("_", "%20")
+            .replace(".", "%20")
+        )
```

Comment:
A regular expression like below may be cleaner:
```suggestion
        url_extension = re.sub(r'[-_. ]', '%20', alarm_name)
```